src/request.js

```diff
@@ -0,0 +1,32 @@
+// Updates the `Request` class to include the ability to manage the `cf`
+// features for the fetch requests.
+//
+// See https://developers.cloudflare.com/workers/reference/cloudflare-features/
+require('whatwg-fetch')
+
+// Available `cf` features. Any keys that are not here and set in `cf`
+// will be discarded.
+const cfFeatures = [
+  "scrapeShield",
+  "polish",
+  "minify",
+  "mirage",
+  "scrapeShield",
+  "apps",
+  "resolveOverride",
+  "cacheKey",
+  "cacheTtl",
+  "cacheTtlByStatus"
+]
+
+const OldRequest = global.Request
+global.Request = class Request extends OldRequest {
+  constructor(input, options) {
+    super(input, options)
+
+    if (options && options.cf) {
+      sanitisedOptions = options.cf.filter(feature => cfFeatures.includes(feature));
```

Comment:
I dont think you can filter a javascript object...

you would need to do something like

const sanitisedFeatures= Object.keys(options.cf).filter(feature => cfFeatures.includes(feature))
const sanitisedOptions = sanitisedFeatures.reduce( //code to build the object, {})